PyRoute/Galaxy.py

```diff
@@ -212,6 +243,21 @@ def __init__(self, name, position):
         self.coreward = None
         self.rimward = None
 
+    def _trim_position(self, head_list, pos_bit, valid_list):
```

Comment:
This seems like an awful lot of work for a pair of integers. I would be reliant on the `int()` method throwing a `ValueError` and a simple check that the two positions were something more than an empty string.

Reply:
Are you sure that will be enough? I've found the following problematic inputs without really trying:
```
    @example('00', '0:,0')
    @example('00', '00,:')
    @example('00', '0A,0')
    @example('00', '0\x1f,0')
    @example('00', '0,0\x1f')
    @example('00', '00,0Ā')
    @example('00', '0-,0')
    @example('00', '0+,0')
    @example('00', '00,-')
    @example('00', '00,+')
    @example('00', '00-,0')
    @example('00', '00,0-')
    @example('00', '0 - 1, -1')
    @example('00', '0 -1, - 1')
    @example('00', '0 - 1, - 1')
    @example('00', '0 ,0')
    @example('00', '00, ')
    @example('0 ', '00,0')
```

And that's without really going nuts - just using the alphabet ```+- 0123456789``` and putting two strings drawn from that alphabet with replacement, flanking a comma.

Reply:
Are these things you have seen in the Traveller Map output or examples you have conjectured could occur?

Reply:
They're examples the hypothesis test generated that _did_ blow up Sector object construction (which was the _point_ of adding the test in the first place) in a way not handled by the code in question.

Reply:
What I'm pointing out here is the T5 sector data files are generated by Traveller Map. Traveller Map doesn't keep the data in this format. So if the data files PyRoute is consuming don't have a reasonable set of position values it means either Traveller Map has a serious new bug, or someone is hand generating a file and mangled it. 

I'm happy to have this in the delta debugging side of the house, I think it is more appropriate there.

Reply:
Assuming these changes belong better in delta debugging, where do you suggest these changes would bolt in?

In your scenario (b0rkated positions in a TravMap download), isn't the attempted run using that mangled file _already_ on a hiding to nowhere?  (Ie, delta debugging benefits from this no matter where it is)

Since the example run is unsalveageable anyway, blow up as loud as possible, as early as possible.  IIRC, sector processing takes a couple of minutes on my current box.

Reply:
I think I figured out why this is weird. I would generally solve this by using a regex, and let the match do the work. 

```
import re
position_match = re.compile(r"([-+ ]?\d{1,4})\s*[,]\s*([-+ ]?\d{1,4})") 
positions = position_match.match(line)
if positions: 
    x = int(positions[1])
    y = int(positions[2])
else:
   # Bad Position code, print error and die
```

This should cover all the cases, rejecting the bad ones, and getting the good ones into something easily found

Reply:
What I pushed out do?